### PR TITLE
JDG-2256 Add REPLICATION_FACTOR template parameter to cache service

### PR DESCRIPTION
**Pull request by galderz**
https://issues.jboss.org/browse/JDG-2256

### CODE REVIEW

**Comment by galderz** [comment]
The following appears in `ScalingTest` when trying to scale up:
```
[2018-11-02T07:30:34Z] [Warning] [Pod:cache-service-1]: 
(FailedScheduling) 0/1 nodes are available: 1 Insufficient cpu.
```

**Comment by galderz** [comment]
@ryanemerson Updated PR. The custom cache service is now created/destroyed in the test, so the cost of the testsuite should remain the same (given that scaling test was already passing before). We do seem however to be close to the limit of the underlying VM for doing integration testing?

File: services/functional-tests/src/test/java/org/infinispan/online/service/utils/CommandLines.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package org.infinispan.online.service.utils;
+
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+
+public class CommandLines {
+
+   private CommandLines() {
```
Comment (ryanemerson) [inline services/functional-tests/src/test/java/org/infinispan/online/service/utils/CommandLines.java]: Unnecessary constructor

File: services/functional-tests/src/test/java/org/infinispan/online/service/utils/CommandLines.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package org.infinispan.online.service.utils;
+
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+
+public class CommandLines {
```
Comment (ryanemerson) [inline services/functional-tests/src/test/java/org/infinispan/online/service/utils/CommandLines.java]: I don't like having two classes with the names `CommandLines` and `CommandLine`. Could we not just add the static methods from `CommandLines` to `CommandLine`?

File: services/functional-tests/src/test/java/org/infinispan/online/service/caching/CustomCachingServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+
+@RunWith(ArquillianConditionalRunner.class)
+@RequiresOpenshift
+public class CustomCachingServiceTest {
```
Comment (ryanemerson) [inline services/functional-tests/src/test/java/org/infinispan/online/service/caching/CustomCachingServiceTest.java:26]: It seems overkill to create a new test just for this. Is it possible to reuse `CachingServiceTest` with a Factory method that provides parameters for each test like we do in https://github.com/infinispan/infinispan? So in this case you would have two tests, one with num_owners=2 and the other with 3 with a generic method `should_have_replication_factor` and then check the configured "numOwners"?

The factory approach will also make it easy to test other startup parameter combinations as well as testing that changing any of those values does not break the caching service behaviour.

**Review by ryanemerson — CHANGES_REQUESTED** [review]

File: services/functional-tests/src/test/java/org/infinispan/online/service/utils/CommandLines.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.infinispan.online.service.utils;
```
Comment (ryanemerson) [inline services/functional-tests/src/test/java/org/infinispan/online/service/utils/CommandLines.java]: @galderz This still needs removing.

**Comment by galderz** [comment]
@ryanemerson Should all be done now

**Pull request merged**